src/components/databrowser/hooks/use-keys.tsx

```diff
@@ -1,61 +1,76 @@
-import {
-  createContext,
-  useCallback,
-  useContext,
-  useMemo,
-  useRef,
-  type PropsWithChildren,
-} from "react"
-import { useDatabrowserStore } from "@/store"
+import { createContext, useContext, useMemo, type PropsWithChildren } from "react"
+import { useDatabrowser, useDatabrowserStore } from "@/store"
+import type { DataType, RedisKey } from "@/types"
 import { useInfiniteQuery, type UseInfiniteQueryResult } from "@tanstack/react-query"
 
 import { useFetchKeyType } from "./use-fetch-key-type"
-import { useFetchKeys, type RedisKey } from "./use-fetch-keys"
 
 const KeysContext = createContext<
   | {
       keys: RedisKey[]
       query: UseInfiniteQueryResult
-      refetch: () => void
     }
   | undefined
 >(undefined)
 
 export const FETCH_KEYS_QUERY_KEY = "use-fetch-keys"
 
+const SCAN_COUNT = 100
+
 export const KeysProvider = ({ children }: PropsWithChildren) => {
   const { search } = useDatabrowserStore()
   const cleanSearchKey = search.key.replace("*", "")
 
   const { data: exactMatchType, isFetching, isLoading } = useFetchKeyType(cleanSearchKey)
-
-  const { fetchKeys, resetCache } = useFetchKeys(search)
-  const pageRef = useRef(0)
+  const { redisNoPipeline: redis } = useDatabrowser()
 
   const query = useInfiniteQuery({
     queryKey: [FETCH_KEYS_QUERY_KEY, search],
 
-    initialPageParam: 0,
-    queryFn: async ({ pageParam: page }) => {
+    initialPageParam: "0",
+    queryFn: async ({ pageParam: lastCursor }) => {
       // We should reset the cache when the pagination is reset
-      if (pageRef.current >= page) resetCache()
-      pageRef.current = page
 
-      return await fetchKeys()
+      const args = [lastCursor]
+
+      if (cleanSearchKey) {
+        args.push("MATCH", cleanSearchKey)
+      }
+
+      if (search.type) {
+        args.push("TYPE", search.type)
+      }
+
+      args.push("COUNT", SCAN_COUNT.toString())
```

Comment:
Noticed an issue with the count parameter. count should be a number, not a string. when it's a string it's value isn't applied.

Removing toString doesn't fix the problem, because exec also stringifyies numbers. It probably shouldn't do that.

Fixing this too ⏳

Reply:
After another look, it wasn't an issue with the parameter being a string. There was another issue.